### PR TITLE
FM2-267: Add create, update and delete methods for ProcedureRequest

### PR DESCRIPTION
**Pull request by varung-31**
## Description of what I changed

Add create, update and delete methods for ProcedureRequest
## Issue I worked on

see https://issues.openmrs.org/browse/FM2-267

## Checklist: I completed these to help reviewers :)

- [x] My IDE is configured to follow the [**code style**](https://wiki.openmrs.org/display/docs/Java+Conventions) of this project.

  No? Unsure? -> [configure your IDE](https://wiki.openmrs.org/display/docs/How-To+Setup+And+Use+Your+IDE), format the code and add the changes with `git add . && git commit --amend`

- [x] I have **added tests** to cover my changes. (If you refactored
  existing code that was well tested you do not have to add tests)

  No? -> write tests and add them to this commit `git add . && git commit --amend`

- [x] I ran `mvn clean package` right before creating this pull request and
  added all formatting changes to my commit.

  No? -> execute above command

- [x] All new and existing **tests passed**.

  No? -> figure out why and add the fix to your commit. It is your responsibility to make sure your code works.

- [x] My pull request is **based on the latest changes** of the master branch.

  No? Unsure? -> execute command `git pull --rebase upstream master`

### CODE REVIEW

**Comment by HerbertYiga** [comment]
cc @ibacher

File: api/src/main/java/org/openmrs/module/fhir2/providers/r3/ProcedureRequestFhirResourceProvider.java [hunk 1]
```diff
@@ -54,6 +63,34 @@ public ProcedureRequest getProcedureRequestById(@IdParam @NotNull IdType id) {
...
+	
+	@Update
+	public MethodOutcome updateProcedureRequest(@IdParam IdType id, @ResourceParam ProcedureRequest procedureRequest) {
+		if (id == null || id.getIdPart() == null) {
+			throw new InvalidRequestException("id must be specified to update");
+		}
+		
+		procedureRequest.setId(id.getIdPart());
+		
+		return FhirProviderUtils.buildUpdate(serviceRequestService.update(id.getIdPart(),
```
Comment (ibacher) [inline api/src/main/java/org/openmrs/module/fhir2/providers/r3/ProcedureRequestFhirResourceProvider.java:80]: This needs to be converted back into a `ProcedureRequest` before it's passed to `buildUpdate()`

File: api/src/main/java/org/openmrs/module/fhir2/providers/r3/ProcedureRequestFhirResourceProvider.java [hunk 2]
```diff
@@ -54,6 +63,34 @@ public ProcedureRequest getProcedureRequestById(@IdParam @NotNull IdType id) {
 		return (ProcedureRequest) VersionConvertor_30_40.convertResource(serviceRequest, false);
 	}
 	
+	@Create
+	public MethodOutcome createProcedureRequest(@ResourceParam ProcedureRequest procedureRequest) {
+		return FhirProviderUtils.buildCreate(
```
Comment (ibacher) [inline api/src/main/java/org/openmrs/module/fhir2/providers/r3/ProcedureRequestFhirResourceProvider.java:68]: This needs to be converted back to a `ProcedureRequest` before it's passed to `buildCreate()`

File: api/src/test/java/org/openmrs/module/fhir2/providers/r3/ProcedureRequestFhirResourceProviderTest.java [hunk 3]
```diff
@@ -107,6 +113,82 @@ public void getServiceRequestByWithWrongId_shouldThrowResourceNotFoundException(
...
+	@Test
+	public void createProcedureRequest_shouldCreateNewProcedureRequest() {
+		when(serviceRequestService.create(any(ServiceRequest.class))).thenReturn(serviceRequest);
+		
+		MethodOutcome result = resourceProvider
+		        .createProcedureRequest((ProcedureRequest) VersionConvertor_30_40.convertResource(serviceRequest, false));
+		
+		assertThat(result, notNullValue());
+		assertThat(result.getCreated(), is(true));
+		assertThat(result.getResource(), equalTo(serviceRequest));
```
Comment (ibacher) [inline api/src/test/java/org/openmrs/module/fhir2/providers/r3/ProcedureRequestFhirResourceProviderTest.java:125]: This is definitely not the expected condition (i.e., an R3 resource provider needs to return the R3 resource).

File: api/src/test/java/org/openmrs/module/fhir2/providers/r3/ProcedureRequestFhirResourceProviderTest.java [hunk 4]
```diff
@@ -107,6 +113,82 @@ public void getServiceRequestByWithWrongId_shouldThrowResourceNotFoundException(
...
+	
+	@Test
+	public void updateProcedureRequest_shouldUpdateProcedureRequest() {
+		when(serviceRequestService.update(eq(SERVICE_REQUEST_UUID), any(ServiceRequest.class))).thenReturn(serviceRequest);
+		
+		MethodOutcome result = resourceProvider.updateProcedureRequest(new IdType().setValue(SERVICE_REQUEST_UUID),
+		    (ProcedureRequest) VersionConvertor_30_40.convertResource(serviceRequest, false));
+		
+		assertThat(result, notNullValue());
+		assertThat(result.getResource(), equalTo(serviceRequest));
```
Comment (ibacher) [inline api/src/test/java/org/openmrs/module/fhir2/providers/r3/ProcedureRequestFhirResourceProviderTest.java:136]: This is definitely not the expected condition (i.e., an R3 resource provider needs to return the R3 resource).

**Comment by varung-31** [comment]
@ibacher Pushed the changes. Please take a look!

**Comment by ibacher** [comment]
@varung-31 Could you resolve the conflicts here?

**Comment by varung-31** [comment]
> @varung-31 Could you resolve the conflicts here?

Done.

**Pull request merged**